app/Http/Controllers/Backend/QsController.php

```diff
@@ -15,64 +13,14 @@ class QsController extends Controller
     public function getBatches(Request $request): \Illuminate\Http\Response
     {
         return DB::transaction(function () {
-            // Get ID of the latest batch that we have created (or 0)
-            $maxIdBatch = QsBatch::orderBy('id', 'desc')->first();
-            if ($maxIdBatch) {
-                $batchesUpToEventId = $maxIdBatch->eventTo;
-            } else {
-                // If there are no batches then things just hav not really started yet
-                $batchesUpToEventId = 0;
-            }
-
-            // Get events after the point that batch was created
-            // TODO maybe filter by NS here?
-            $events = EventPageUpdate::where('id', '>', $batchesUpToEventId)->get();
-            $wikiBatchesEntities = [];
-            $lastEventId = 0;
-            foreach ($events as $event) {
-                if (
-                    $event->namespace == 120 || // item
-                    $event->namespace == 122 || // property
-                    $event->namespace == 146// lexeme
-                ) {
-                    $wikiBatchesEntities[$event->wiki_id][] = $event->title;
-                }
-                if ($event->id > $lastEventId) {
-                    $lastEventId = $event->id;
-                }
-            }
-
-            /** @var Collection $newlyCreatedBatches */
-            $newlyCreatedBatches = [];
-            $notDoneBatches = QsBatch::where([
-                ['done', '=', 0], ['pending_since', '=', null], ['failed', '=', false]
+            $oldestBatch = QsBatch::where([
+                ['done', '=', 0],
+                ['pending_since', '=', null],
+                ['failed', '=', false]
             ])
-                ->with(['wiki', 'wiki.wikiQueryserviceNamespace'])->get();
-            // Inset the newly created batches into the table...
-            foreach ($wikiBatchesEntities as $wikiId => $entityBatch) {
-
-                // If we already have a not done batch for this same wiki, then merge that into a new batch
-                foreach ($notDoneBatches as $qsBatch) {
-                    if ($qsBatch->wiki_id == $wikiId) {
-                        $entityBatch = array_merge($entityBatch, explode(',', $qsBatch->entityIds));
-                        // Delete the old batch
-                        $qsBatch->delete();
-                    }
-                }
-
-                // Insert the new batch
-                $batch = QsBatch::create([
-                    'done' => 0,
-                    'eventFrom' => $batchesUpToEventId,
-                    'eventTo'=> $lastEventId,
-                    'wiki_id' => $wikiId,
-                    'entityIds' => implode(',', array_unique($entityBatch)),
-                    'pending_since' => null,
-                ]);
-                $newlyCreatedBatches[] = $batch;
-            }
+                ->orderBy('id')
```

Comment:
let's double check this doesn't need a `desc`

Reply:
Double checked: It's fine as is.